src/js/schemas/service-schema/HealthChecks.js

```diff
@@ -0,0 +1,132 @@
+import FormUtil from '../../utils/FormUtil';
+
+let HealthChecks = {
+  type: 'object',
+  title: 'Health Checks',
+  properties: {
+    healthChecks: {
+      type: 'array',
+      duplicable: true,
+      addLabel: 'Add Another Health Check',
+      getter: function (service) {
+        let healthChecks = service.getHealthChecks();
+        if (healthChecks == null) {
+          return [];
+        }
+
+        return healthChecks;
+      },
+      deleteButtonTop: true,
+      filterProperties: function (service = {}, instanceDefinition) {
+        let properties = HealthChecks
+          .properties
+          .healthChecks
+          .itemShape
+          .properties;
+
+        instanceDefinition.forEach(function (definition) {
+          let prop = definition.name;
+          if (FormUtil.isFieldInstanceOfProp('healthChecks', definition)) {
+            prop = FormUtil.getPropKey(definition.name);
+          }
+          if (properties[prop].shouldShow) {
+            definition.formElementClass = {
+              'hidden-form-element': !properties[prop].shouldShow(service)
```

Comment:
nice, this is an interesting way of doing it. i like it. one problem with this is that if you choose protocol HTTP, type in a path, then choose protocol COMMAND, the path value will still be there if you switch to JSON mode.

a solution to this could be to do cleanup of the unnecessary props when switching